core/domain/summary_services.py

```diff
@@ -339,6 +451,22 @@ def _get_displayable_collection_summary_dicts(collection_summaries):
 def get_library_groups(language_codes):
     """Returns a list of groups for the library index page. Each group has a
     header and a list of dicts representing activity summaries.
+
+    Args:
+        language_codes: list(str). A list of language code
+
+    Return:
+        list(dict). List of groups(dict); for each group there is a list of
```

Comment:
```
list(dict). A list of groups for the library index page. Each group is represented by a dict with the following keys and values:
    - activity_summary_dicts: ...
    - categories: ...
    ...
```

where the "..." should be replaced by mini-docstrings explaining clearly the type and meaning of each key-value pair.